### PR TITLE
Add support for SushiSwap LP tokens

### PR DESCRIPTION
**Pull request by julienmartinlevrai**

### CODE REVIEW

**Review by talbaneth — COMMENTED** [review]
Probably need to add to the readme and emphasize that when using sushi lp the deployer needs to call getOrCreateProxy() once before he can take(). Was considering doing it in take() itself but that's probably too gas wasteful. Also the constructor is not a good option since this can be deployed independetly/before the manager. So I think it is good as is, but best adding to the readme.

**Comment by julienmartinlevrai** [comment]
@hexonaut I am unable to add you as a reviewer, for some reason

File: README.md [hunk 1]
```diff
@@ -23,10 +23,11 @@ Each exchange has its own public callee contract, which exposes a function calle
 NOTE:
 * NEVER DIRECTLY SEND internal/ERC20 `gem`/`dai` to an exchange callee contract.
 * Remember to call `vat.hope(clipper)` from the msg.sender once before calling `Clipper.take`
+* For Sushi LP tokens and other forms of collateral that use CropManager, you need to create a Proxy for the address where you are receiving your funds by calling `CropManager.getOrCreateProxy()` once.
 
 ## Exchanges supported
 * [OasisDex](https://oasisdex.com/)
-* [UniswapV2](https://uniswap.org/) (Only ETH collateral types are supported at the moment)
+* [UniswapV2](https://uniswap.org/)
```
Comment (talbaneth) [inline README.md:30]: should we add sushiswap?

**Comment by talbaneth** [comment]
Reviewed the contract change (not the tests yet).
Looks good.

Few notes:
* For sushi lp this will need a different deployment (since the router is different).
* For sushi lp deployment we currently have the same router contract for both swapping and removing liquidity, so we can only swap in sushi.

I think the above points are fine, especially as this a sample code.

**Review by hexonaut — APPROVED** [review]
LGTM 👍

**Review by talbaneth — COMMENTED** [review]
looks good, see minor comments

File: src/test/Simulation.t.sol [hunk 2]
```diff
@@ -22,8 +22,11 @@ import "dss-interfaces/Interfaces.sol";
 import { UniswapV2CalleeDai } from "../UniswapV2Callee.sol";
 import { UniswapV2LpTokenCalleeDai } from "../UniswapV2LpTokenCallee.sol";
 
-import "dss/clip.sol";
-import "dss/abaci.sol";
+import { CropManager, CropManagerImp } from "dss-crop-join/CropManager.sol";
+import { SushiJoin } from "dss-crop-join/SushiJoinV2.sol";
+import { ProxyManagerClipper } from "dss-crop-join/ProxyManagerClipper.sol";
```
[the remark this answers is not in the document]
Comment (julienmartinlevrai) [reply src/test/Simulation.t.sol:27]: Is it private? I don’t have access to that.

[the remark this answers is not in the document]
Comment (talbaneth) [reply src/test/Simulation.t.sol:27]: probably, @gbalabasquer should be able to add you.

File: src/test/Simulation.t.sol [hunk 3]
```diff
@@ -190,21 +230,73 @@ contract SimulationTests is DSTest {
...
+
+    function deployContracts() private {
+        sushiManagerImp = new CropManagerImp(vatAddr);
+        sushiManagerImpAddr = address(sushiManagerImp);
+        sushiManager = new CropManager();
+        sushiManagerAddr = address(sushiManager);
+        sushiManager.setImplementation(sushiManagerImpAddr);
+        CropManagerLike(sushiManagerAddr).getOrCreateProxy(danAddr);
+        uint256 pid = 0;
+        address rewarder = 0x7519C93fC5073E15d89131fD38118D73A72370F8;
```
[the remark this answers is not in the document]
Comment (julienmartinlevrai) [reply src/test/Simulation.t.sol:253]: They are only used for the context of this function. Moving them would require creating contract fields.

**Comment by godsflaw** [comment]
Closing this as Sushi rugged Maker on the integration ages ago.

**Pull request closed without merging**